frontend/model/contracts/voting/rules.js

```diff
@@ -15,14 +15,22 @@ export const RULE_MULTI_CHOICE = 'multi-choice'
 
 // TODO: ranked-choice? :D
 
+/* REVIVEW PR
+  the whole "state" being passed as argument to rules[rule]() is overwhelmed.
+  It would be simpler with just 2 simple args: "threshold" and "population".
+  Advantages:
+  - population: No need to do the logic to get it (and avoid import PROFILE_STATUS.ACTIVE from group.js).
+  - threshold: The selector to get the selector is huge.
+  - tests: Avoid the need to mock a complex state.
+  My suggestion: 1 parameter (object) with 4 explicit keys.
+  [RULE_PERCENTAGE]: function ({ votes, proposalType, population, threshold })
```

Comment:
We keep it the way it is currently because you cannot know what you might need in the future for different rule types.

Reply:
I understand that. When that happens in the future, why can't we just adapt the code to pass a new parameter?

At the moment, as I see it, we are passing down the whole jungle when we just need 4 bananas.

Reply:
Besides that, when passing `state`, we need `PROFILE_STATUS` to get `population`. When importing it we create a `circular dependency`.
> Circular dependency: frontend/model/contracts/group.js -> frontend/model/contracts/voting/rules.js -> frontend/model/contracts/group.js

Reply:
Update: This took me a long time, but unit tests were failing exactly because of this circular dependency. (too complex to explain in a short comment)

Anyway, I added a const `PROFILE_STATUS` to this file as a short solution until you get back to this PR :)

Reply:
I gave this some thought, and think we can resolve it by doing the following:

1. Move the file `model/constants.js` into `model/contracts/contants.js`
2. Move the constants `INVITE_INITIAL_CREATOR`, `INVITE_STATUS`, and `PROFILE_STATUS` in `group.js` into `model/contracts/contants.js`
3. Then in `rules.js` and `group.js`, import those constants as needed.
4. Remove the comment above, since it's part of the design to pass `state` into the function. If you're worried about performance, don't be; it shouldn't have any effect on performance since JavaScript passes values by reference.

Reply:
Yes, keeping constants in its own file is much better. Changed